### DISCUSSION TITLE
Upstream changes for pandoc codefence parsing

### DISCUSSION THREAD

**Discussion by dmadisetti** [quarto-dev/quarto-cli#9835] [Feature Requests]
### Description

I opened this issue in pandoc for handling naked attributes as empty attributes in https://github.com/jgm/pandoc/issues/9791

That means Pandoc will consider

````md
```{attr}
//
```
````

as

````md
```{attr=""}
//
```
````

Behind the scenes.

As far as I can tell, this won't change quarto behavior- but wanted to raise it here if there are concerns.

**Comment by cscheid** [comment]
Thanks for letting us know, we'll keep an eye on it.

FYI, Quarto already needs to do a complicated dance around

````
```{lang}
...
```
````

because plain Pandoc 3 doesn't recognize that markdown syntax as a code block (and Quarto needs to, because that's our syntax for block-level code cells). Quarto uses a custom reader for that, and that custom reader carefully converts `{lang}` into an equivalent code block that Pandoc _does_ parse. I believe that custom reader means we won't be affected by the incoming upstream change, whatever it turns out to be. 

The relevant lines are:

https://github.com/quarto-dev/quarto-cli/blob/7506cb699cf87d6cbb85559e91ccddce6f17fc57/src/resources/pandoc/datadir/readqmd.lua#L65-L113

**Reply by dmadisetti** [reply]
Oh woops, I was looking at `escape_engine`. I see. I guess `escape_engine` escapes engine notation, not extract it. Thanks!
